app/assets/javascripts/wpcc/components/UpdateResults.js

```diff
@@ -0,0 +1,96 @@
+  'use strict';
+
+  var UpdateResults,
+    defaultConfig = {};
+
+  UpdateResults = function($el, config) {
+    UpdateResults.baseConstructor.call(this, $el, config, defaultConfig);
+
+    this.frequencySelector = this.$el.find('[data-dough-selector]');
+    this.resultsTables = this.$el.find('[data-dough-results-table]')
+    this.values = {
+      employeeContributions: [],
+      employerContributions: [],
+      taxRelief: []
+    };
+  };
+
+  /**
+  * Inherit from base module, for shared methods and interface
+  */
+  DoughBaseComponent.extend(UpdateResults);
+
+  UpdateResults.componentName = 'UpdateResults';
+
+  UpdateResults.prototype.init = function(initialised) {
+    this._initialisedSuccess(initialised);
+    this._setupEvents();
+    this._getValues();
+  };
+
+  // Set event for change of frequency selector
+  UpdateResults.prototype._setupEvents = function() {
+    var _this = this;
+
+    this.frequencySelector.change(function() {
+      _this._displayValues(this.value);
+    });
+  }
+
+  // Get the intial values from the DOM
+  // and store in values object
+  UpdateResults.prototype._getValues = function() {
+    var _this = this;
+    var frequency = this.frequencySelector.val();
+    var modifier = this._getModifier(frequency);
+
+    this.resultsTables.each(function() {
+      _this.values.employeeContributions.push($(this).find('[data-dough-employee-contribution]').attr('data-value') * modifier);
+      _this.values.employerContributions.push($(this).find('[data-dough-employer-contribution]').attr('data-value') * modifier);
+      _this.values.taxRelief.push($(this).find('[data-dough-tax-relief]').attr('data-value') * modifier);
+    });
+  }
+
+  // Display recalculated values in tables
+  UpdateResults.prototype._displayValues = function(frequency) {
+    var _this = this;
+    var modifier = this._getModifier(frequency);
+
+    // calculate & display new values
+    for (var i = 0, max = this.resultsTables.length; i < max; i++) {
+      $(this.resultsTables[i]).find('[data-dough-employee-contribution]').html(_this._formatNumber(_this.values.employeeContributions[i], modifier));
+      $(this.resultsTables[i]).find('[data-dough-employer-contribution]').html(_this._formatNumber(_this.values.employerContributions[i], modifier));
+      $(this.resultsTables[i]).find('[data-dough-tax-relief]').html('(includes tax relief of ' + _this._formatNumber(_this.values.taxRelief[i], modifier) + ')');
+      $(this.resultsTables[i]).find('[data-dough-total]').html(_this._formatNumber(_this.values.employeeContributions[i] + _this.values.employerContributions[i], modifier));
+    };
+  }
+
+  // Get modifier from frequency
+  // Used to convert amounts from/to annual
+  UpdateResults.prototype._getModifier = function(frequency) {
+    switch(frequency) {
+      case 'month':
+        return 12;
+      case 'fourweeks':
+        return 13;
+      case 'week':
+        return 52;
+    }
+
+    return 1;
+  }
```

Comment:
Hmm, yes that might be more elegant in fact, though that's not native for JS, but I'll have a quick look. I'm not unhappy with the method name since it does just return a value that is used in the calculations so seems descriptive enough.